activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1111,7 +1111,7 @@ def does_not_support_reverse?(order)
         # Uses SQL function with multiple arguments.
         (order.include?(",") && order.split(",").find { |section| section.count("(") != section.count(")") }) ||
           # Uses "nulls first" like construction.
-          /nulls (first|last)\Z/i.match?(order)
+          /nulls (first|last)/i.match?(order)
```

Comment:
If the `order` has two or more space between `nulls` and `first`, the regex still doesn't match to the `order`.
(e.g. `Topic.order(Arel.sql("title NULLS   FIRST"))`)

```suggestion
          /\bnulls\s+(?:first|last)\b/i.match?(order)
```